etw/logrus/hook.go

```diff
@@ -0,0 +1,78 @@
+package hook

```

Comment:
Is a nested package really the right approach here? Why not just have this in the etw package?

Reply:
I wanted to have the etw package just be the base support for ETW itself, and put things building on top of it in other packages. Seemed like good separation, and ensures the etw package exposes the right set of things to allow something meaningful to be built with it.

Reply:
I guess what I am saying is that people don't write packages in go like:
```
pkg\datatypes
pkg\datatypes\mytype.go
pkg\datatypes\json
pkg\datatypes\json\mytype_json.go
pkg\datatypes\toml
pkg\datatypes\toml\mytype_toml.go
```
And in the `json` and `toml` files add the serializers.

The would write
```
pkg\datatypes
pkg\datatypes\mytype.go
pkg\datatypes\mytype_json.go
pkg\datatypes\mytype_toml.go
```
And then the caller can decide which ones to include.

So for this it seems:
```
pkg\etw
pkg\etw\stuff.go
pkg\etw\logrus_hook.go
pkg\etw\myother_hook.go
```
OR Two packages
```
pkg\etw
pkg\etwlogrus
```

Reply:
If I go with the first approach, they'll have to all be in the same package, right? I'd rather ensure that the ETW Logrus hook can't use non-exported things from ETW, so I think they should be in different packages.

The second approach looks fine to me, though.